third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -279,6 +279,11 @@ pub enum Type {
         ty_args: TriompheArc<Vec<Type>>,
         ability: AbilityInfo,
     },
+    Function {
+        args: Vec<TriompheArc<Type>>,
```

Comment:
Why `Vec<TriompheArc<Type>>` and not `TriompheArc<Vec<Type>>`? having each element as an arcs adds quite a bit of extra memory allocations, and we don't really share any element types - all types are read-only right now. From perf perspective, I think arced vector makes more sense.

Reply:
Good point! I think they are only used to avoid deep clones that you would do with a box, but otherwise these should be 1 at all times I believe.